crates/bitcoin-da/src/helpers/parsers.rs

```diff
@@ -45,22 +86,94 @@ impl ParsedInscription {
     }
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+impl VerifyParsed for ParsedComplete {
+    fn public_key(&self) -> &[u8] {
+        &self.public_key
+    }
+    fn signature(&self) -> &[u8] {
+        &self.signature
+    }
+    fn body(&self) -> &[u8] {
+        &self.body
+    }
+}
+
+impl VerifyParsed for ParsedAggregate {
+    fn public_key(&self) -> &[u8] {
+        &self.public_key
+    }
+    fn signature(&self) -> &[u8] {
+        &self.signature
+    }
+    fn body(&self) -> &[u8] {
+        &self.body
+    }
+}
+
+impl VerifyParsed for ParsedSequencerCommitment {
+    fn public_key(&self) -> &[u8] {
+        &self.public_key
+    }
+    fn signature(&self) -> &[u8] {
+        &self.signature
+    }
+    fn body(&self) -> &[u8] {
+        &self.body
+    }
+}
+
+impl ParsedAggregate {
+    pub fn txids(&self) -> Result<Vec<Txid>, bitcoin::hashes::FromSliceError> {
+        self.body.chunks_exact(32).map(Txid::from_slice).collect()
+    }
+}
+
+#[derive(Error, Debug, Clone, PartialEq)]
 pub enum ParserError {
+    #[error("Invalid rollup name")]
     InvalidRollupName,
-    EnvelopeHasNonPushOp,
-    EnvelopeHasIncorrectFormat,
+    #[error("Invalid header length")]
+    InvalidHeaderLength,
+    #[error("Invalid header type {0}")]
+    InvalidHeaderType(NonZeroU16),
+    #[error("No witness in tapscript")]
     NonTapscriptWitness,
-    IncorrectSignature,
+    #[error("Unexpected end of script")]
+    UnexpectedEndOfScript,
+    #[error("Invalid opcode in the script")]
+    UnexpectedOpcode,
+    #[error("Script error: {0}")]
+    ScriptError(String),
+}
+
+impl From<ScriptError> for ParserError {
+    fn from(value: ScriptError) -> ParserError {
+        ParserError::ScriptError(value.to_string())
+    }
 }
 
 pub fn parse_transaction(
     tx: &Transaction,
     rollup_name: &str,
-) -> Result<ParsedInscription, ParserError> {
+) -> Result<ParsedLightClientTransaction, ParserError> {
+    let script = get_script(tx)?;
+    let instructions = script.instructions().peekable();
+    // Map all Instructions errors into ParserError::ScriptError
+    let mut instructions = instructions.map(|r| r.map_err(ParserError::from));
```

Comment:
name it parse_light_client_transaction -- or something else if we come up with another naming (see other comments)